datafusion/execution/src/cache/cache_unit.rs

```diff
@@ -94,9 +94,69 @@ impl CacheAccessor<Path, Arc<Statistics>> for DefaultFileStatisticsCache {
     }
 }
 
+/// Collected files metadata for listing files.
+/// Cache will not invalided until user call remove or clear.
```

Comment:
I think here keep simple, user can use their own notify system to decide call `remove` or `clear`